### PR TITLE
ETK: update Jetpack codesniffer to 2.6

### PR DESCRIPTION
**Pull request by simison**
#### Proposed Changes

* Updates Jetpack codesniffer from 2.1 to 2.6. See [changelog](https://github.com/Automattic/jetpack/blob/59d5eb5a97365d555ae8c8d6647a95700e941f39/projects/packages/codesniffer/CHANGELOG.md#261---2022-11-01)
* Latest version adds new check for script registrations to include `text_domain`. Adds configuration for it.
* Ignored few more rules for the `newspack-blocks` folder which is git-ignored and the code is managed elsewhere 
* Run `yarn lint:php:fix` with results:
```
PHPCBF RESULT SUMMARY
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------
FILE                                                                                                                                                      FIXED  REMAINING
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------
editing-toolkit/editing-toolkit-plugin/whats-new/class-whats-new.php                                                                                      1      0
editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles.php                                                                              1      0
editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/class-wpcom-block-editor-nux.php                                                            1      0
editing-toolkit/editing-toolkit-plugin/help-center/class-help-center.php                                                                                  1      0
editing-toolkit/editing-toolkit-plugin/wpcom-documentation-links/class-wpcom-documentation-links.php                                                      1      0
editing-toolkit/editing-toolkit-plugin/tags-education/class-tags-education.php                                                                            1      0
editing-toolkit/editing-toolkit-plugin/wpcom-block-description-links/class-wpcom-block-description-links.php                                              1      0
editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nav-sidebar/class-wpcom-block-editor-nav-sidebar.php                                            1      0
```

#### Testing Instructions

* In `apps/editing-toolkit` run `yarn lint:php`

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #

### CODE REVIEW

File: apps/editing-toolkit/editing-toolkit-plugin/coming-soon/coming-soon.php [hunk 1]
```diff
@@ -20,7 +20,7 @@ function should_show_coming_soon_page() {
 	$share_code = get_share_code();
 	if ( is_accessed_by_valid_share_link( $share_code ) ) {
 		track_preview_link_event();
-		setcookie( 'wp_share_code', $share_code, time() + 3600, '/', false, is_ssl() );
+		setcookie( 'wp_share_code', $share_code, time() + 3600, '/', false, is_ssl(), true );
```
Comment (simison) [inline apps/editing-toolkit/editing-toolkit-plugin/coming-soon/coming-soon.php:23]: Also fixed in the PR moving this code to Jetpack repo: https://github.com/Automattic/jetpack/pull/27815/files#diff-fdd367b51a6fa0f82a7ec1faa3c7751756f98acd665f37618bde58e7766adb00R23

File: apps/phpcs.xml [hunk 2]
```diff
@@ -29,9 +38,15 @@
 		ignore style rules not tracked upstream. (We should not ignore rules
 		which catch actual errors.)
 	-->
+	<rule ref="Squiz.Commenting.FunctionComment.MissingParamTag" >
```
Comment (simison) [inline apps/phpcs.xml:49]: Alternatively we could ignore the entire directory for all rules.

**Comment by simison** [comment]
CI is failing with:

> Error: The PHPUnit Polyfills library is a requirement for running the WP test suite.
If you are trying to run plugin/theme integration tests, make sure the PHPUnit Polyfills library (https://github.com/Yoast/PHPUnit-Polyfills) is available and either load the autoload file of this library in your own test bootstrap before calling the WP Core test bootstrap file; or set the absolute path to the PHPUnit Polyfills library in a "WP_TESTS_PHPUNIT_POLYFILLS_PATH" constant to allow the WP Core bootstrap to load the Polyfills.
>
> If you are trying to run the WP Core tests, make sure to set the "WP_RUN_CORE_TESTS" constant to 1 and run `composer update -W` before running the tests.
Once the dependencies are installed, you can run the tests using the Composer-installed version of PHPUnit or using a PHPUnit phar file, but the dependencies do need to be installed whichever way the tests are run.

Needs a closer look.

**Edit:** looks like it's failing in other PRs as well, e.g. https://github.com/Automattic/wp-calypso/pull/71051

**Comment by noahtallen** [comment]
wp-env was updated with a new way to handle phpunit some months ago. I started this PR to bring it into ETK a while ago: #65653. Once we switch to that approach, I think this error would go away.

But I'm also not sure why the changes in this PR start introducing that error!

I also don't think I'll be able to finish that before leaving on holidays for a couple weeks, unfortunately.

**Pull request closed without merging**